### PR TITLE
Create developer documentation.

### PR DESCRIPTION
**Pull request by gnecula**
* Moved out of README.md some developer-only stuff to docs/developer.rst.
    * Added documentation about building the documentation

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -149,82 +149,6 @@ if you run into any errors or problems with the prebuilt wheels.
 
 ### Building JAX from source
```
Comment (hawkinsp) [inline README.md]: This section header now seems misleading.

In addition, some users will need to build from source, so even if the build instructions move elsewhere, we would still want a link here I think.

File: docs/developer.rst [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+`pytest <https://docs.pytest.org/en/latest/usage.html#specifying-tests-selecting-tests>`_'s
+built-in selection mechanisms, or alternatively you can run a specific test
+file directly to see more detailed information about the cases being run:
+
+.. code-block:: shell
+
+ python tests/lax_numpy_test.py --num_generated_cases=5
+
+
+Update documentation
```
Comment (hawkinsp) [inline docs/developer.rst]: There is a docs/README.md that probably should link here.

File: README.md (hunk 1 above)
Comment (gnecula) [reply README.md]: Done

File: docs/developer.rst (hunk 2 above)
Comment (gnecula) [reply docs/developer.rst]: I added the link, but I debated even removing the docs/README.md. I left it in for now.

**Review by hawkinsp — APPROVED** [review]

**Pull request merged**